src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,21 @@
+package duke.task;
+
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String writeToFile(){
```

Comment:
Good use of override to allow you to write to file effectively (in terms of writing the code). It is easy to read and understand too.